app/src/main/java/com/mensinator/app/business/ClueImport.kt

```diff
@@ -0,0 +1,106 @@
+package com.mensinator.app.business
+
+import android.content.Context
+import android.widget.Toast
+import org.json.JSONArray
+import java.io.BufferedReader
+import java.io.File
+import java.io.InputStreamReader
+import java.time.LocalDate
+import java.time.format.DateTimeFormatter
+import androidx.core.database.sqlite.transaction
+import java.io.FileInputStream
+
+class ClueImport(
+    private val context: Context,
+    private val dbHelper: IPeriodDatabaseHelper,
+) : IClueImport {
+
+    override fun getDefaultImportFilePath(): String {
+        return File(context.getExternalFilesDir(null), "import.json").absolutePath
+    }
+
+    override fun importFileToDatabase(filePath: String) {
+        val db = dbHelper.writableDb
+
+        // Read JSON data from the file
+        val file = File(filePath)
+        val fileInputStream = FileInputStream(file)
+        val reader = BufferedReader(InputStreamReader(fileInputStream))
+
+        Toast.makeText(context, "Importing file...", Toast.LENGTH_SHORT).show()
+
+        val stringBuilder = StringBuilder()
+        var line: String? = reader.readLine()
+        while (line != null) {
+            stringBuilder.append(line)
+            line = reader.readLine()
+        }
+        reader.close()
+
+        // Parse JSON array from the file
+        val importArray = JSONArray(stringBuilder.toString())
+
+        // Validate the data before cleanup
+        if (!validateImportData(importArray)) {
+            Toast.makeText(context, "Invalid data in import file", Toast.LENGTH_SHORT).show()
+            return
+        }
+
+        val formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd")
+
+        db.transaction {
+            try {
+                for (i in 0 until importArray.length()) {
+                    val obj = importArray.getJSONObject(i)
+                    val type = obj.getString("type")
+                    val dateString = obj.getString("date")
+                    val date = LocalDate.parse(dateString, formatter)
+
+                    if (type == "period") {
```

Comment:
Replace by when condition (you can ask Android Studio to do this) to improve readability